htsohm/pseudomaterial_generator/random.py

```diff
@@ -0,0 +1,138 @@
+from random import random
+
+from Crypto.Hash import SHA256
+from zlib import crc32
+
+from htsohm.db import Material
+from htsohm.structure import Structure, LatticeConstants, AtomSite, AtomType
+
+# random number generator
+def bytes_to_float(b):
+    return float(crc32(b) & 0xffffffff) / 2**32
+
+def str_to_float(s, encoding="utf-8"):
+    return bytes_to_float(s.encode(encoding))
+
+def random_number(n):
+    # use SHA256 to get random number from some input value
+    h = SHA256.new('{}'.format(n).encode())
+    return str_to_float(h.hexdigest())
+
+# uniform selection from random number
+def uniform_selection(a, b, r):
+    return b - r * (b - a)
+
+def random_number_density(number_density_limits, structure, r): #lattice_constants):
+    """Produces random number for atom-sites in a unit cell, constrained by
+    some number density limits.
+
+    Args:
+        number_density_limits (list): min and max number densities, for example:
+            [min_(float), max_(float)]
+        lattice_constants (dict): crystal lattice constants, for example:
+            {"a" : (float),
+             "b" : (float),
+             "c" : (float)}
+    
+    Returns:
+        atoms (int): some random number of atom-sites under the imposed limits.
+
+        If the minimum number density results in a unit cell with less than 2
+        atom-sites with the given lattice constants, a minimum number density
+        of TWO ATOM SITES PER UNIT CELL is imposed.
+    
+    """
+    min_ND = number_density_limits[0]
+    max_ND = number_density_limits[1]
+    v = structure.volume()
+    min_atoms = int(min_ND * v)
```

Comment:
use math.ceil instead of int

Reply:
commit 71417d407148f441b739ed1a1e9c497b4ebf5948
Author: akaija <a.r.kaija@gmail.com>
Date:   Wed Jun 20 16:09:46 2018 -0400

    fixed random number density function